src/modules/user/user.module.ts

```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { UserRepositoryImpl } from './user.repository';
+import { UserService } from './user.service';
+import { UserController } from './user.controller';
+
+@Module({
+  controllers: [UserController],
+  providers: [
+    UserService,
+    { provide: 'UserRepository', useClass: UserRepositoryImpl },
```

Comment:
This should only be UserRepository, no need for abstractions at this point. Make sure the "repository" is injectable.